src/bucket.ts

```diff
@@ -3174,6 +3185,201 @@ class Bucket extends ServiceObject {
+   *     `options.metadata.contentEncoding` to `gzip`.
+   * @param {object} [options.metadata] See an
+   *     [Objects: insert request
+   * body](https://cloud.google.com/storage/docs/json_api/v1/objects/insert#request_properties_JSON).
+   * @param {string} [options.offset] The starting byte of the upload stream, for
+   *     resuming an interrupted upload. Defaults to 0.
+   * @param {string} [options.predefinedAcl] Apply a predefined set of access
+   *     controls to every object.
+   *
+   *     Acceptable values are:
+   *     - **`authenticatedRead`** - Object owner gets `OWNER` access, and
+   *       `allAuthenticatedUsers` get `READER` access.
+   *
+   *     - **`bucketOwnerFullControl`** - Object owner gets `OWNER` access, and
+   *       project team owners get `OWNER` access.
+   *
+   *     - **`bucketOwnerRead`** - Object owner gets `OWNER` access, and project
+   *       team owners get `READER` access.
+   *
+   *     - **`private`** - Object owner gets `OWNER` access.
+   *
+   *     - **`projectPrivate`** - Object owner gets `OWNER` access, and project
+   *       team members get access according to their roles.
+   *
+   *     - **`publicRead`** - Object owner gets `OWNER` access, and `allUsers`
+   * get `READER` access.
+   * @param {boolean} [options.private] Make the uploaded files private. (Alias for
+   *     `options.predefinedAcl = 'private'`)
+   * @param {boolean} [options.public] Make the uploaded files public. (Alias for
+   *     `options.predefinedAcl = 'publicRead'`)
+   * @param {boolean} [options.resumable] Force a resumable upload. (default:
+   *     true for files larger than 5 MB).
+   * @param {string} [options.uri] The URI for an already-created resumable
+   *     upload. See {@link File#createResumableUpload}.
+   * @param {string} [options.userProject] The ID of the project which will be
+   *     billed for the request.
+   * @param {string|boolean} [options.validation] Possible values: `"md5"`,
+   *     `"crc32c"`, or `false`. By default, data integrity is validated with an
+   *     MD5 checksum for maximum reliability. CRC32c will provide better
+   *     performance with less reliability. You may also choose to skip
+   * validation completely, however this is **not recommended**.
+   */
+  /**
+   * @typedef {array} UploadDirectoryResponse
+   * @property {object} [response] Response for each of the uploaded files.
+   * @param {string} [filename] Absolute file path.
+   * @param {string | Error} [status] Status of the file upload. "success"
+   *     or an Error returned while attempting to upload current file.
+   */
+  /**
+   * @callback UploadDirectoryCallback
+   * @param {?Error} err Request error, if any.
+   * @param {array} resp The list of object responses for every uploaded file.
+   */
+
+  /**
+   * Upload a directory to the bucket. This is a convenience method for multiple
+   * {@link Bucket#upload}.
```

Comment:
I think maybe this should say something like
> This is a convenience method for making multiple `{@link Bucket#upload}` calls.

Reply:
done